### PR TITLE
Add a new PCOM layer

### PR DESCRIPTION
**Pull request by lmrosa**
new layer for PCOM SCADA protocol

### CODE REVIEW

**Comment by lmrosa** [comment]
@gpotter2 is it supposed to have all the SCADA related protocols in a dedicated folder
[https://github.com/secdev/scapy/pull/1241#discussion_r265207221](https://github.com/secdev/scapy/pull/1241#discussion_r265207221) ? 

If so, I think several protocols should be moved as well, all of them are SCADA related:
[Modbus](https://github.com/secdev/scapy/blob/master/scapy/contrib/modbus.py), [ENIP](https://github.com/secdev/scapy/blob/master/scapy/contrib/enipTCP.py), [Profinet](https://github.com/secdev/scapy/blob/master/scapy/contrib/pnio.py) and this [https://github.com/secdev/scapy/pull/1898](https://github.com/secdev/scapy/pull/1898) of course.

Anyway any feedback on this one ?

File: scapy/contrib/pcom.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        PCOMAsciiCommandField(
+            "command", '', length_from=lambda pkt: pkt.underlayer.len - 7),
+        XShortField("chksum", None),
+        XByteField("etx", 0x0d)
+    ]
+
+    def post_build(self, pkt, pay):
+        if self.chksum is None:
+            chksum = PCOMAscii.pcom_ascii_checksum(pkt[2:-3])
+            pkt = pkt[:-3] + pack("2B", chksum[0], chksum[1]) + pkt[-1:]
```
Comment (gpotter2) [inline scapy/contrib/pcom.py]: If I'm not mistaken, all the calls to `pcom_ascii_checksum` are followed by the same `struct.pack` call. The latter could be moved directly in `pcom_ascii_checksum`

**Comment by gpotter2** [comment]
> If so, I think several protocols should be moved as well, all of them are SCADA related:
Modbus, ENIP, Profinet

Those protocols have been implemented for too long to allow us to move them freely. If you move them, add an alias for them in `Scapy/contrib/__init__`. This PR should however be included there.

**Comment by guedou** [comment]
PR #2371 includes the fixes requested by @gpotter2.

**Pull request closed without merging**